src/postman/operation.ts

```diff
@@ -26,18 +32,37 @@ export const transformPostmanCollectionOperation: PostmanCollectionHttpOperation
     throw new Error(`Unable to find "${method} ${path}"`);
   }
 
+  const auth = item.getAuth();
+  const postmanSecurity = auth && findPostmanSecurityScheme(auth, collection);
+  const request = transformRequest(item.request);
+  const security: HttpSecurityScheme[][] = [];
+
+  if (postmanSecurity) {
+    switch (postmanSecurity.type) {
+      case 'securityScheme':
+        security.push([postmanSecurity.securityScheme]);
+        break;
+      case 'headerParams':
+        request.headers!.push(...postmanSecurity.headerParams);
```

Comment:
You got me. It will be always set because of how `transformRequest` above works. Meaning, it always returns an array.

But you right, I should act like I don posses that knowledge and verify the existence of `headers` and `query`.

Reply:
> You got me. It will be always set because of how transformRequest above works. Meaning, it always returns an array.

Hmm, if `transformRequest` gives you these guarantees, we can just modify it to return the correct type, with the property marked as `NonNullable`

Reply:
https://github.com/piotrwitek/utility-types#requiredt-k

Reply:
I tried using the built-in utility types and I failed. These fill some gaps, thanks!

Reply:
That's fine, these are just types so they do not even get into the final bundle — augmenting the size or something